### ISSUE TITLE
Request failed: poloniex GET https://poloniex.com/public?command=returnTicker 403 Forbidden

### ISSUE THREAD

**Issue by trungtv** [enigmampc/catalyst#161]
Dear Catalyst Maintainers,
It seems that poloniex block non-human request from my computer. I can never run the following tutorial.
https://enigmampc.github.io/catalyst/beginner-tutorial.html#ingesting-data
Is there a way not to connect to Poloniex by default or, do I always need an API Key from poloniex even thought I don't plan to do live trading?

Thank you very much, 
T

**Comment by lacabra** [comment]
If you are outside the United States, Poloniex requires captcha for all API requests, thereby rendering their API unusable. Use any of the two other exchanges instead: `bitfinex` or `bittrex`

The section in the tutorial you refer to, though, explicitly uses `bitfinex` to avoid this problem. Thus, I am bit confused by the specific challenge that you are experiencing. If this doesn't solve your issue, please include the specific command that you're trying to run, and the full stack traceback of your error.

**Comment by trungtv** [comment]
Dear Lacabra, 
Thanks for your prompt answer. I tried again to extract market data
[https://enigmampc.github.io/catalyst/utilities.html#extracting-market-data](url)
```
results = run_algorithm(initialize=initialize,
                        handle_data=handle_data,
                        analyze=analyze,
                        start=start,
                        end=end,
                        exchange_name='bitfinex',
                        data_frequency='minute',
                        base_currency ='usdt',
                        capital_base=10000 )
```
I changed exchange_name = 'bitfinex' as suggested but catalyst engine still unexpectedly connect to poloniex somehow...

**Comment by trungtv** [comment]
Sorry, 
I forgot to attached the full stack trackback
```
---------------------------------------------------------------------------
ExchangeRequestError                      Traceback (most recent call last)
<ipython-input-3-264afa5a29c8> in <module>()
     44                         data_frequency='minute',
     45                         base_currency ='usdt',
---> 46                         capital_base=10000 )

/Users/trungtv/anaconda3/envs/catalyst/lib/python2.7/site-packages/catalyst/utils/run_algo.pyc in run_algorithm(initialize, capital_base, start, end, handle_data, before_trading_start, analyze, data_frequency, data, bundle, bundle_timestamp, default_extension, extensions, strict_extensions, environ, live, exchange_name, base_currency, algo_namespace, live_graph, simulate_orders, stats_output, output)
    561         live_graph=live_graph,
    562         simulate_orders=simulate_orders,
--> 563         stats_output=stats_output
    564     )

/Users/trungtv/anaconda3/envs/catalyst/lib/python2.7/site-packages/catalyst/utils/run_algo.pyc in _run(handle_data, initialize, before_trading_start, analyze, algofile, algotext, defines, data_frequency, capital_base, data, bundle, bundle_timestamp, start, end, output, print_algo, local_namespace, environ, live, exchange, algo_namespace, base_currency, live_graph, simulate_orders, stats_output)
    171         environ=environ,
    172         exchange_tz='UTC',
--> 173         asset_db_path=None  # We don't need an asset db, we have exchanges
    174     )
    175     env.asset_finder = AssetFinderExchange()

/Users/trungtv/anaconda3/envs/catalyst/lib/python2.7/site-packages/catalyst/finance/trading.pyc in __init__(self, load, bm_symbol, exchange_tz, trading_calendar, asset_db_path, future_chain_predicates, environ)
     99             trading_calendar.day,
    100             trading_calendar.schedule.index,
--> 101             self.bm_symbol,
    102         )
    103 

/Users/trungtv/anaconda3/envs/catalyst/lib/python2.7/site-packages/catalyst/data/loader.pyc in load_crypto_market_data(trading_day, trading_days, bm_symbol, bundle, bundle_data, environ, exchange, start_dt, end_dt)
    146         from catalyst.exchange.factory import get_exchange
    147         exchange = get_exchange(
--> 148             exchange_name='poloniex', base_currency='usdt'
    149         )
    150 

/Users/trungtv/anaconda3/envs/catalyst/lib/python2.7/site-packages/catalyst/exchange/factory.pyc in get_exchange(exchange_name, base_currency, must_authenticate)
     23         key=exchange_auth['key'],
     24         secret=exchange_auth['secret'],
---> 25         base_currency=base_currency,
     26     )
     27 

/Users/trungtv/anaconda3/envs/catalyst/lib/python2.7/site-packages/catalyst/exchange/ccxt/ccxt_exchange.pyc in __init__(self, exchange_name, key, secret, base_currency)
     64 
     65         except ExchangeNotAvailable as e:
---> 66             raise ExchangeRequestError(error=e)
     67 
     68         self.name = exchange_name

ExchangeRequestError: Request failed: poloniex GET https://poloniex.com/public?command=returnTicker 403 Forbidden  <!DOCTYPE html>
<html lang="en">
<head>
```

**Comment by BalthazarBratt** [comment]
I'm having the same issue. Changing exchange_name doesn't have any impact.

**Comment by cbogithub** [comment]
same issue, too. i wish it will solve asap

**Comment by fredfortier** [comment]
I believe that this is the same issue: https://github.com/ccxt/ccxt/issues/994

I don't believe that it is an application issue per se, but CCXT has a proxy workaround which we haven't considered but should enable. You should get the same error when reaching Polo from a web browser from the same computer, which might make it easier to find a solution.

**Comment by lacabra** [comment]
We are using Poloniex to fetch `btc_usdt` and use it as a benchmark to calculate performance statistics for the algos. Seeing how Poloniex is problematic for so many people, an interim solution is to fetch the benchmark from another exchange. Currently working to resolve this issue.

**Comment by fredfortier** [comment]
Fixed on 0.4.6 please re-open if you still notice an issue.
